test/integration/deployment/deployment_test.go

```diff
@@ -1403,3 +1407,425 @@ func TestTerminatingReplicasDeploymentStatus(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestRecreateDeploymentForPodReplacement(t *testing.T) {
```

Comment:
These tests will include [additional test cases](https://github.com/kubernetes/kubernetes/pull/123430/commits/2631590ceb22e0e01d5ab9c3fc42e76408522e59) once we add the PodReplacementPolicy